ethereum/Makefile

```diff
@@ -0,0 +1,32 @@
+SOURCE_FILES:=$(shell find contracts -name "*.sol")
+
+.PHONY: dependencies test clean all
+
+all: build
+
+node_modules: package-lock.json
+	touch -m node_modules
```

Comment:
Is it possible for this to run before the `node_modules` directory exists? If so, it will create a file, which I'm assuming `npm ci` will not be terribly happy about.

Reply:
It is very much possible, but npm actually removes the file in that case. Not the cleanest though, should we `mkdir -p node_modules` first?

Reply:
That would be better than `touch node_modules`, yes.